test/e2e/common/node/pod_level_resources_resize.go

```diff
@@ -0,0 +1,710 @@
+		resources.Limits[v1.ResourceCPU] = resource.MustParse(cpuLim)
+	}
+
+	if memLim != "" {
+		resources.Limits[v1.ResourceMemory] = resource.MustParse(memLim)
+
+	}
+
+	if cpuReq != "" || memReq != "" {
+		resources.Requests = v1.ResourceList{}
+	}
+
+	if cpuReq != "" {
+		resources.Requests[v1.ResourceCPU] = resource.MustParse(cpuReq)
+	}
+
+	if memReq != "" {
+		resources.Requests[v1.ResourceMemory] = resource.MustParse(memReq)
+
+	}
+	return resources
+}
+
+func VerifyPodLevelStatus(gotPod *v1.Pod) error {
+	var errs []error
+	var wantStatusResources *v1.ResourceRequirements
+
+	if gotPod.Status.Phase != v1.PodRunning {
+		wantStatusResources = gotPod.Spec.Resources
+		if err := framework.Gomega().Expect(gotPod.Status.Resources).To(gomega.BeComparableTo(wantStatusResources)); err != nil {
+			errs = append(errs, fmt.Errorf("pod.status.resources mismatch: %w", err))
+		}
+	}
+	// TODO(ndixita) - add tests when to compare Pod.Status.Resources with pod
+	// cgroup config values. Converting cgroup values -> Pod.Status.Resources ->
+	// cgroup values is resulting in indeterministic rounded off values
+
+	var wantAllocatedResources v1.ResourceList
+	aggrReq, aggrLim := podresize.AggregateContainerResources(gotPod.Spec)
+	wantStatusResources = &v1.ResourceRequirements{Requests: aggrReq, Limits: aggrLim}
+	wantAllocatedResources = aggrReq
+
+	if gotPod.Spec.Resources != nil {
+		wantStatusResources = &v1.ResourceRequirements{
+			Requests: gotPod.Spec.Resources.Requests,
+			Limits:   gotPod.Spec.Resources.Limits,
+		}
+		wantAllocatedResources = gotPod.Spec.Resources.Requests
+	}
+
+	if err := framework.Gomega().Expect(gotPod.Status.AllocatedResources).To(gomega.BeComparableTo(wantAllocatedResources)); err != nil {
+		errs = append(errs, fmt.Errorf("pod.status.allocatedResources mismatch: %w", err))
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func doPodLevelResourcesMemoryLimitDecreaseTest(f *framework.Framework) {
+	// Tests the behavior when decreasing pod-level memory limit:
+	// 1. Decrease the limit a little bit - should succeed
+	// 2. Decrease the limit down to a tiny amount - should fail
+	// 3. Revert the limit back to the original value - should succeed
+	ginkgo.It("decrease pod-level memory limit below usage", func(ctx context.Context) {
+		podClient := e2epod.NewPodClient(f)
+		originalPLR := &v1.ResourceRequirements{
+			Requests: v1.ResourceList{
+				v1.ResourceMemory: resource.MustParse(originalMem),
+			},
+			Limits: v1.ResourceList{
+				v1.ResourceMemory: resource.MustParse(originalMem),
+			},
+		}
+
+		containers := []podresize.ResizableContainerInfo{{
+			Name: "c1",
+		}}
+
+		ginkgo.By("creating and verifying pod")
+		testPod := createAndVerifyPodPLR(ctx, f, podClient, containers, originalPLR, true)
+
+		// 1. Decrease the limit a little bit - should succeed
+		ginkgo.By("Patching pod with a slightly lowered memory limit")
+		viableLoweredLimit := []podresize.ResizableContainerInfo{{
+			Name: "c1",
+		}}
```

Comment:
nit: remove this. there shouldn't be any container-level resizes, so just reuse the original `containers`